zap/src/output/luau/server.rs

```diff
@@ -250,10 +254,96 @@ impl<'a> ServerOutput<'a> {
 			self.push_stmts(statements);
 		}
 
-		if ev.call == EvCall::SingleSync || ev.call == EvCall::SingleAsync {
-			self.push_line(&format!("if reliable_events[{id}] then"))
-		} else {
-			self.push_line(&format!("for _, cb in reliable_events[{id}] do"))
+		match ev.call {
+			EvCall::SingleSync | EvCall::SingleAsync => self.push_line(&format!("if reliable_events[{id}] then")),
+			EvCall::ManySync | EvCall::ManyAsync => self.push_line(&format!("for _, cb in reliable_events[{id}] do")),
+			EvCall::Polling => {
```

Comment:
A lot of the same comments as `client.rs`, so please make sure they're applied in this file as well